src/sankoff_handler.cpp

```diff
@@ -53,115 +57,92 @@ EigenVectorXd SankoffHandler::ParentPartial(EigenVectorXd child_partials) {
   return parent_partials;
 }
 
-EigenVectorXd SankoffHandler::TotalPPartial(SankoffPartial left, SankoffPartial right,
-                                            size_t node_id) {
-  return left.block<1, state_count_>(node_id, 0) +
-         right.block<1, state_count_>(node_id, 0);
+EigenVectorXd SankoffHandler::TotalPPartial(size_t node_id, size_t site_idx) {
+  return 
+  psv_handler_.GetPV(PSVType::PLeft, node_id).block<state_count_, 1>(0, site_idx)+
+  psv_handler_.GetPV(PSVType::PRight, node_id).block<state_count_, 1>(0, site_idx);
 }
 
-EigenVectorXd SankoffHandler::TotalPPartial(SankoffPartial all_partials_for_site,
-                                            size_t node_count, size_t node_id) {
-  return all_partials_for_site.block<1, state_count_>(
-             GetSankoffPartialIndex(SankoffPartialType::PLeft, node_count, node_id),
-             0) +
-         all_partials_for_site.block<1, state_count_>(
-             GetSankoffPartialIndex(SankoffPartialType::PRight, node_count, node_id),
-             0);
-}
 
-SankoffPartialVec SankoffHandler::BlankPartials(size_t node_count) {
-  SankoffPartialVec blank_partials(site_pattern_.PatternCount());
-  for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
-       pattern_idx++) {
-    SankoffPartial blank_partials_per_site(node_count, state_count_);
-    blank_partials_per_site.setZero();
-    blank_partials[pattern_idx] = blank_partials_per_site;
-  }
-  return blank_partials;
-}
+void SankoffHandler::RunSankoff(Node::NodePtr topology) {
+  // make sure that the psv_handler is sized to fit the current topology
```

Comment:
I'm not sure if a comment is needed here, but if you want one let's make a full grammatical sentence, please.

Reply:
resolved